apps/financial-aid/api/src/app/modules/decorators/currentUser.decorator.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  decodeToken,
+  IDENTITY_SERVER_SESSION_TOKEN_COOKIE_NAME,
+  User,
+} from '@island.is/financial-aid/shared/lib'
+
+import {
+  createParamDecorator,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common'
+
+import { GqlExecutionContext } from '@nestjs/graphql'
+
+export const CurrentUser = createParamDecorator(
+  (_: unknown, context: ExecutionContext): User => {
+    const req = GqlExecutionContext.create(context).getContext().req
+
+    const sessionToken = req.cookies
+      ? req.cookies[IDENTITY_SERVER_SESSION_TOKEN_COOKIE_NAME]
+      : null
+
+    if (!sessionToken) {
+      throw new UnauthorizedException('Invalid user')
```

Comment:
This is a decorator for GraphQL and your errorLink in the front-end is performing actions based on error extension code: `err.extensions?.code`. I don't think that `UnauthorizedException` creates those error codes for you.

I think the correct approach here is to throw the `AuthenticationError` from 'apollo-server-express' which is the server that @nestjs/graphql builds upon.
That one comes with error codes which your errorlink understands.